internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -1166,6 +1167,23 @@ func (r *BMACReconciler) ensureMCSCert(ctx context.Context, log logrus.FieldLogg
 		bmh.ObjectMeta.Annotations[BMH_AGENT_IGNITION_CONFIG_OVERRIDES] = ignitionWithMCSCert
 	}
 	log.Info("MCS certificate injected")
+
+	clusterInstall := hiveext.AgentClusterInstall{}
+	namespacedName := types.NamespacedName{
+		Namespace: cd.Namespace,
+		Name:      cd.Spec.ClusterInstallRef.Name,
+	}
+	if err = r.Client.Get(ctx, namespacedName, &clusterInstall); err != nil {
+		return reconcileError{errors.Wrapf(err, "Could not get AgentClusterInstall %s for ClusterDeployment %s", cd.Spec.ClusterInstallRef.Name, cd.Name)}
+	}
+	clusterInstall.Spec.IgnitionEndpoint = &hiveext.IgnitionEndpoint{
+		Url:                    fmt.Sprintf("https://%s:22623/config/%s", clusterInstall.Status.APIVIP, agent.Status.Role),
```

Comment:
Certificate is per ip? Not dns name?

Reply:
We assumed that it is per dns name. We have field apidnsname for day2 cluster. Maybe better to use it?